tracing-serde/src/lib.rs

```diff
@@ -15,6 +15,36 @@ use tracing_core::{
     span::{Attributes, Id, Record},
 };
 
+/// A bridge between `fmt::Write` and `io::Write`.
+///
+/// This is needed because tracing-subscriber's FormatEvent expects a fmt::Write
+/// while serde_json's Serializer expects an io::Write.
+pub struct WriteAdaptor<'a> {
+    fmt_write: &'a mut dyn fmt::Write,
+}
+
+impl<'a> WriteAdaptor<'a> {
+    pub fn new(fmt_write: &'a mut dyn fmt::Write) -> Self {
+        Self { fmt_write }
+    }
+}
+
+impl<'a> io::Write for WriteAdaptor<'a> {
+    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+        let s = String::from_utf8_lossy(buf);
```

Comment:
IMO this should probably be
```suggestion
        let s = String::from_utf8(buf)?;
```
and return an error if there are non-utf8 characters in the buffer...

Reply:
Also, this could use [`str::from_utf8`](https://doc.rust-lang.org/std/str/fn.from_utf8.html) and not have to allocate.
```suggestion
        let s = std::str::from_utf8(buf)?;
```

Reply:
Why I didn't find this before... Thought it's only for `String`...